### ISSUE TITLE
ValueError and StopIteration in search

### ISSUE THREAD

**Issue by mytluo** [dib-lab/sourmash#522]
Hi everyone,
I'm going through the sourmash tutorial in the documentation, but every time I try to run sourmash search I get this error:

> Error in parsing signature; quitting.
> Exception: generator raised StopIteration
> Traceback (most recent call last):
>   File "/Users/melissa/py3/lib/python3.7/site-packages/sourmash/signature_json.py", line 188, in load_signatureset_json_iter
>     ijson=ijson)
>   File "/Users/melissa/py3/lib/python3.7/site-packages/sourmash/signature_json.py", line 128, in load_signature_json
>     raise ValueError('expected "start_map".')
> ValueError: expected "start_map".
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/Users/melissa/py3/lib/python3.7/site-packages/sourmash/signature_json.py", line 193, in load_signatureset_json_iter
>     prefix, event, value = next(parser)
> StopIteration
> 
> The above exception was the direct cause of the following exception:
> 
> Traceback (most recent call last):
>   File "/Users/melissa/py3/bin/sourmash", line 11, in <module>
>     load_entry_point('sourmash==2.0.0a9', 'console_scripts', 'sourmash')()
>   File "/Users/melissa/py3/lib/python3.7/site-packages/sourmash/__main__.py", line 77, in main
>     cmd(sys.argv[2:])
>   File "/Users/melissa/py3/lib/python3.7/site-packages/sourmash/commands.py", line 761, in search
>     select_moltype=moltype)
>   File "/Users/melissa/py3/lib/python3.7/site-packages/sourmash/sourmash_args.py", line 86, in load_query_signature
>     sl = list(sl)
>   File "/Users/melissa/py3/lib/python3.7/site-packages/sourmash/signature.py", line 213, in load_signatures
>     ignore_md5sum=ignore_md5sum):
>   File "/Users/melissa/py3/lib/python3.7/site-packages/sourmash/signature_json.py", line 217, in load_signatures_json
>     for n, sigset in enumerate(it):
> RuntimeError: generator raised StopIteration

This happened when I tried to run
`sourmash search -k 31 ecoli-reads.sig ecoli-genome.sig --containment`
as well as 
`sourmash search ecoli-genome.sig ecolidb.sbt.json -n 20`

Is there something I'm doing wrong? 

Thank you,
Melissa

**Comment by ctb** [comment]
Hi Melissa, that is very weird! It looks like you're going through
the tutorial, is that correct?

**Comment by mytluo** [comment]
Hi @ctb, yep, I'm doing the tutorial, which is why I'm not sure where the error is coming from :(

**Comment by ctb** [comment]
I think I figured it out --

instead of:

cd ~/sourmash
sourmash compute --scaled 10000 ~/data/ecoli_ref*pe*.fq.gz -o ecoli-reads.sig -k 31

can you try running:

cd ~/sourmash
ourmash compute --scaled 10000 ~/data/ecoli_ref-5m.fastq.gz -o ecoli-reads.sig -k 31

and then continue the tutorial from

Compare reads to assemblies


onwards?

sorry, and thanks --

--titus

On Tue, Jul 24, 2018 at 11:19:47AM -0700, mytluo wrote:
> Hi @ctb, yep, I'm doing the tutorial, which is why I'm not sure where the error is coming from :( 
> 
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/dib-lab/sourmash/issues/522#issuecomment-407503753
-- 
C. Titus Brown, ctbrown@ucdavis.edu

**Comment by mytluo** [comment]
@ctb, I just tried it but sadly the error is still there. The StopIteration error also pops up when I try to run sourmash compare:

```
(py3) Melissa-Luos-MacBook-Pro13:sourmash melissa$ sourmash compare ecoli*.sig -o cmp
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
loaded 4 signatures total.
downsampling to scaled value of 10000

0-ecoliMG1655.fa.gz 	[1. 0. 0. 0.]
1-ecoli_ref-5m.fa...	[0.106 1.    0.    0.   ]
2-ecoliMG1655.fa.gz 	[1.    0.106 1.    0.   ]
3-ecoli_ref-5m.fa...	[0.106 1.    0.106 1.   ]
min similarity in matrix: 0.106
saving labels to: cmp.labels.txt
saving distance matrix to: cmp
```

**Comment by mytluo** [comment]
I'm wondering if maybe there has been a change in the files themselves that are causing the error? I'm trying the sourmash LCA tutorial as well, and when I get to the step to build the LCA database, I get the same error:

```
$ sourmash lca index tara-delmont-SuppTable3.csv delmont.lca.json delmont-subsample-sigs/*.sig
examining spreadsheet headers...
** assuming column 'MAGs' is identifiers in spreadsheet
** assuming column 'Domain' is superkingdom in spreadsheet
106 distinct lineages in spreadsheet out of 957 rows
finding signatures...
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
Error in parsing signature; quitting.
Exception: generator raised StopIteration
...found 100 genomes with lineage assignments!!
37 assigned lineages out of 106 distinct lineages in spreadsheet
saving to LCA DB: delmont.lca.json
WARNING: no signatures for 857 lineage assignments.
WARNING: 69 unused lineages.
(You can use --report to generate a detailed report.)
```

**Comment by mytluo** [comment]
Though when I go to the next step to run lca classify, it (kind of) works:

```
$ sourmash lca classify --db delmont.lca.json --query delmont-subsample-sigs/TARA_RED_MAG_00003.fa.gz.sig
loaded 1 LCA databases. ksize=31, scaled=10000
finding query signatures...
outputting classifications to stdout
ID,status,superkingdom,phylum,class,order,family,genus,species,strain
TARA_RED_MAG_00003,found,Bacteria,Proteobacteria,Gammaproteobacteria,,,,,
Error in parsing signature; quitting.
Exception: generator raised StopIteration
classified 1 signatures total
```

Except for the Error message, the output is as expected according to the tutorial...

**Comment by ctb** [comment]
yes, I think you have a bad signature that's causing the problems -
if you can 'rm *.sig' and then regenerate, I wonder if the error
messages would go away?

On Tue, Jul 24, 2018 at 08:37:57PM +0000, mytluo wrote:
> Though when I go to the next step to run lca classify, it (kind of) works:
> 
> ```
> $ sourmash lca classify --db delmont.lca.json --query delmont-subsample-sigs/TARA_RED_MAG_00003.fa.gz.sig
> loaded 1 LCA databases. ksize=31, scaled=10000
> finding query signatures...
> outputting classifications to stdout
> ID,status,superkingdom,phylum,class,order,family,genus,species,strain
> TARA_RED_MAG_00003,found,Bacteria,Proteobacteria,Gammaproteobacteria,,,,,
> Error in parsing signature; quitting.
> Exception: generator raised StopIteration
> classified 1 signatures total
> ```
> 
> Except for the Error message, the output is as expected according to the tutorial...
> 
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/dib-lab/sourmash/issues/522#issuecomment-407543879
-- 
C. Titus Brown, ctbrown@ucdavis.edu

**Comment by mytluo** [comment]
I've regenerated the signature files about half a dozen times (both with --scaled 10000 and --scaled 1000), as well as tried redownloading the original data files (to confirm, the reference genome file is 432.5 MB and the MG1655 genome is 1.4 MB?, with the signature files being 8 KB for the reference and 77 KB for the reads), with the same error messages. The `compute` function seems to run fine though:

```
$ sourmash compute --scaled 10000 ecoli_ref-5m.fastq.gz -o ecoli-reads.sig -k 31
setting num_hashes to 0 because --scaled is set
computing signatures for files: ecoli_ref-5m.fastq.gz
Computing signature for ksizes: [31]
Computing only DNA (and not protein) signatures.
Computing a total of 1 signature(s).
... reading sequences from ecoli_ref-5m.fastq.gz
calculated 1 signatures for 5000000 sequences in ecoli_ref-5m.fastq.gz
saved 1 signature(s). Note: signature license is CC0.
```

```
$ sourmash compute --scaled 10000 ecoliMG1655.fa.gz -o ecoli-genome.sig -k 31
setting num_hashes to 0 because --scaled is set
computing signatures for files: ecoliMG1655.fa.gz
Computing signature for ksizes: [31]
Computing only DNA (and not protein) signatures.
Computing a total of 1 signature(s).
... reading sequences from ecoliMG1655.fa.gz
calculated 1 signatures for 1 sequences in ecoliMG1655.fa.gz
saved 1 signature(s). Note: signature license is CC0.
```

**Comment by mytluo** [comment]
@ctb, apologies again but I just came across another error when trying to do the tutorial for "What's in my metagenome?"

```
$ sourmash gather -k 31 shakya-unaligned-contigs.sig genbank-k31.sbt.json
Error in parsing signature; quitting.
Exception: Unexpected symbol '<' at 0
Traceback (most recent call last):
  File "/Users/melissa/py3/lib/python3.7/site-packages/ijson/backends/python.py", line 125, in parse_value
    yield ('number', common.number(symbol))
  File "/Users/melissa/py3/lib/python3.7/site-packages/ijson/common.py", line 158, in number
    number = decimal.Decimal(str_value)
decimal.InvalidOperation: [<class 'decimal.ConversionSyntax'>]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/melissa/py3/bin/sourmash", line 11, in <module>
    load_entry_point('sourmash==2.0.0a9', 'console_scripts', 'sourmash')()
  File "/Users/melissa/py3/lib/python3.7/site-packages/sourmash/__main__.py", line 77, in main
    cmd(sys.argv[2:])
  File "/Users/melissa/py3/lib/python3.7/site-packages/sourmash/commands.py", line 940, in gather
    select_moltype=moltype)
  File "/Users/melissa/py3/lib/python3.7/site-packages/sourmash/sourmash_args.py", line 86, in load_query_signature
    sl = list(sl)
  File "/Users/melissa/py3/lib/python3.7/site-packages/sourmash/signature.py", line 213, in load_signatures
    ignore_md5sum=ignore_md5sum):
  File "/Users/melissa/py3/lib/python3.7/site-packages/sourmash/signature_json.py", line 217, in load_signatures_json
    for n, sigset in enumerate(it):
  File "/Users/melissa/py3/lib/python3.7/site-packages/sourmash/signature_json.py", line 179, in load_signatureset_json_iter
    prefix, event, value = next(parser)
  File "/Users/melissa/py3/lib/python3.7/site-packages/ijson/common.py", line 65, in parse
    for event, value in basic_events:
  File "/Users/melissa/py3/lib/python3.7/site-packages/ijson/backends/python.py", line 185, in basic_parse
    for value in parse_value(lexer):
  File "/Users/melissa/py3/lib/python3.7/site-packages/ijson/backends/python.py", line 127, in parse_value
    raise UnexpectedSymbol(symbol, pos)
ijson.backends.python.UnexpectedSymbol: Unexpected symbol '<' at 0
(py3) Melissa-Luos-MacBook-Pro13:sourmash melissa$ sourmash gather -k 31 shakya-unaligned-contigs.sig genbank-k31.sbt.json
Error in parsing signature; quitting.
Exception: Unexpected symbol '<' at 0
Traceback (most recent call last):
  File "/Users/melissa/py3/lib/python3.7/site-packages/ijson/backends/python.py", line 125, in parse_value
    yield ('number', common.number(symbol))
  File "/Users/melissa/py3/lib/python3.7/site-packages/ijson/common.py", line 158, in number
    number = decimal.Decimal(str_value)
decimal.InvalidOperation: [<class 'decimal.ConversionSyntax'>]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/melissa/py3/bin/sourmash", line 11, in <module>
    load_entry_point('sourmash==2.0.0a9', 'console_scripts', 'sourmash')()
  File "/Users/melissa/py3/lib/python3.7/site-packages/sourmash/__main__.py", line 77, in main
    cmd(sys.argv[2:])
  File "/Users/melissa/py3/lib/python3.7/site-packages/sourmash/commands.py", line 940, in gather
    select_moltype=moltype)
  File "/Users/melissa/py3/lib/python3.7/site-packages/sourmash/sourmash_args.py", line 86, in load_query_signature
    sl = list(sl)
  File "/Users/melissa/py3/lib/python3.7/site-packages/sourmash/signature.py", line 213, in load_signatures
    ignore_md5sum=ignore_md5sum):
  File "/Users/melissa/py3/lib/python3.7/site-packages/sourmash/signature_json.py", line 217, in load_signatures_json
    for n, sigset in enumerate(it):
  File "/Users/melissa/py3/lib/python3.7/site-packages/sourmash/signature_json.py", line 179, in load_signatureset_json_iter
    prefix, event, value = next(parser)
  File "/Users/melissa/py3/lib/python3.7/site-packages/ijson/common.py", line 65, in parse
    for event, value in basic_events:
  File "/Users/melissa/py3/lib/python3.7/site-packages/ijson/backends/python.py", line 185, in basic_parse
    for value in parse_value(lexer):
  File "/Users/melissa/py3/lib/python3.7/site-packages/ijson/backends/python.py", line 127, in parse_value
    raise UnexpectedSymbol(symbol, pos)
ijson.backends.python.UnexpectedSymbol: Unexpected symbol '<' at 0
```

On a different note, I still get the StopIteration runtime error when using different genome files, so I'm not sure if that is related.

**Comment by ctb** [comment]
keep 'em coming -- this is important stuff :) I am traveling at the moment
but will get back to you shortly.

best,
--titus

On Wed, Jul 25, 2018 at 01:52:40PM -0700, mytluo wrote:
> @ctb, apologies again but I just came across another error when trying to do the tutorial for "What's in my metagenome?"
> 
> ```
> $ sourmash gather -k 31 shakya-unaligned-contigs.sig genbank-k31.sbt.json
> Error in parsing signature; quitting.
> Exception: Unexpected symbol '<' at 0
> Traceback (most recent call last):
>   File "/Users/melissa/py3/lib/python3.7/site-packages/ijson/backends/python.py", line 125, in parse_value
>     yield ('number', common.number(symbol))
>   File "/Users/melissa/py3/lib/python3.7/site-packages/ijson/common.py", line 158, in number
>     number = decimal.Decimal(str_value)
> decimal.InvalidOperation: [<class 'decimal.ConversionSyntax'>]
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/Users/melissa/py3/bin/sourmash", line 11, in <module>
>     load_entry_point('sourmash==2.0.0a9', 'console_scripts', 'sourmash')()
>   File "/Users/melissa/py3/lib/python3.7/site-packages/sourmash/__main__.py", line 77, in main
>     cmd(sys.argv[2:])
>   File "/Users/melissa/py3/lib/python3.7/site-packages/sourmash/commands.py", line 940, in gather
>     select_moltype=moltype)
>   File "/Users/melissa/py3/lib/python3.7/site-packages/sourmash/sourmash_args.py", line 86, in load_query_signature
>     sl = list(sl)
>   File "/Users/melissa/py3/lib/python3.7/site-packages/sourmash/signature.py", line 213, in load_signatures
>     ignore_md5sum=ignore_md5sum):
>   File "/Users/melissa/py3/lib/python3.7/site-packages/sourmash/signature_json.py", line 217, in load_signatures_json
>     for n, sigset in enumerate(it):
>   File "/Users/melissa/py3/lib/python3.7/site-packages/sourmash/signature_json.py", line 179, in load_signatureset_json_iter
>     prefix, event, value = next(parser)
>   File "/Users/melissa/py3/lib/python3.7/site-packages/ijson/common.py", line 65, in parse
>     for event, value in basic_events:
>   File "/Users/melissa/py3/lib/python3.7/site-packages/ijson/backends/python.py", line 185, in basic_parse
>     for value in parse_value(lexer):
>   File "/Users/melissa/py3/lib/python3.7/site-packages/ijson/backends/python.py", line 127, in parse_value
>     raise UnexpectedSymbol(symbol, pos)
> ijson.backends.python.UnexpectedSymbol: Unexpected symbol '<' at 0
> (py3) Melissa-Luos-MacBook-Pro13:sourmash melissa$ sourmash gather -k 31 shakya-unaligned-contigs.sig genbank-k31.sbt.json
> Error in parsing signature; quitting.
> Exception: Unexpected symbol '<' at 0
> Traceback (most recent call last):
>   File "/Users/melissa/py3/lib/python3.7/site-packages/ijson/backends/python.py", line 125, in parse_value
>     yield ('number', common.number(symbol))
>   File "/Users/melissa/py3/lib/python3.7/site-packages/ijson/common.py", line 158, in number
>     number = decimal.Decimal(str_value)
> decimal.InvalidOperation: [<class 'decimal.ConversionSyntax'>]
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/Users/melissa/py3/bin/sourmash", line 11, in <module>
>     load_entry_point('sourmash==2.0.0a9', 'console_scripts', 'sourmash')()
>   File "/Users/melissa/py3/lib/python3.7/site-packages/sourmash/__main__.py", line 77, in main
>     cmd(sys.argv[2:])
>   File "/Users/melissa/py3/lib/python3.7/site-packages/sourmash/commands.py", line 940, in gather
>     select_moltype=moltype)
>   File "/Users/melissa/py3/lib/python3.7/site-packages/sourmash/sourmash_args.py", line 86, in load_query_signature
>     sl = list(sl)
>   File "/Users/melissa/py3/lib/python3.7/site-packages/sourmash/signature.py", line 213, in load_signatures
>     ignore_md5sum=ignore_md5sum):
>   File "/Users/melissa/py3/lib/python3.7/site-packages/sourmash/signature_json.py", line 217, in load_signatures_json
>     for n, sigset in enumerate(it):
>   File "/Users/melissa/py3/lib/python3.7/site-packages/sourmash/signature_json.py", line 179, in load_signatureset_json_iter
>     prefix, event, value = next(parser)
>   File "/Users/melissa/py3/lib/python3.7/site-packages/ijson/common.py", line 65, in parse
>     for event, value in basic_events:
>   File "/Users/melissa/py3/lib/python3.7/site-packages/ijson/backends/python.py", line 185, in basic_parse
>     for value in parse_value(lexer):
>   File "/Users/melissa/py3/lib/python3.7/site-packages/ijson/backends/python.py", line 127, in parse_value
>     raise UnexpectedSymbol(symbol, pos)
> ijson.backends.python.UnexpectedSymbol: Unexpected symbol '<' at 0
> ```
> 
> On a different note, I still get the StopIteration runtime error when using different genome files, so I'm not sure if that is related.
> 
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/dib-lab/sourmash/issues/522#issuecomment-407891601
-- 
C. Titus Brown, ctbrown@ucdavis.edu

**Comment by mytluo** [comment]
Hi @ctb, just wanted to update you -- I do not get these errors when running sourmash in a EC2 instance. It works great! However, I do get a difference in output for one of the steps from tutorial:

```sourmash search -k 31 ecolireads.sig ecoligenome.sig --containment
loaded query: ecoli_ref-5m.fastq.gz... (k=31, DNA)
loaded 1 signatures.

1 matches:
similarity   match
----------   -----
  9.7%       ecoliMG1655.fa.gz
```
expected from tutorial:

> # running sourmash subcommand: search
> loaded query: /home/ubuntu/data/ecoli_ref-5m... (k=31, DNA)
> loaded 1 signatures from ecoli-genome.sig
> 1 matches:
> similarity   match
> ----------   -----
>  46.6%       /home/ubuntu/data/ecoliMG1655.fa.gz

**Comment by luizirber** [comment]
So, the `StopIteration` error is coming from Python 3.7 and `ijson` interactions, I'm also seeing this but haven't figured out why it happens. It still works on Python 2.7 or 3.5-3.6.

**Comment by mytluo** [comment]
@luizirber ahh that actually makes sense. When I first installed sourmash locally on my computer I hadn't specified Python 3.5 when setting up the venv; but in the EC2 instance I did, which explains why it works in the instance but not locally for me.

**Comment by standage** [comment]
I'm running the tutorial on CentOS 7 and ran into the same `sourmash search` discrepancy that @mytluo describes above: 10.6% rather than 46.6%.

Also in the next step (searching the SBT) I get the following error.

```
$ sourmash search ecoli-genome.sig ecolidb.sbt.json -n 20
select query k=31 automatically.
loaded query: /home/standaged/data/ecoliMG1655.... (k=31, DNA)
for tree 'ecolidb.sbt.json', scaled value is smaller than query.
tree scaled: 2000; query scaled: 10000. Cannot do similarity search.
```

The tutorial specifies running `sourmash compute` with `--scaled 10000`. Is this conflicting with the default `--scaled` value for `sourmash index`? Has this changed since the tutorial was written?

**Comment by standage** [comment]
I should clarify I'm running Python 3.6.

**Comment by luizirber** [comment]
Fixed by #546, I'll release `2.0.0a10` to include it
